extensions/amp-timeago/0.2/timeago.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElement, useRef, useState} from '../../../src/preact';
+import {timeago} from '../../../third_party/timeagojs/timeago';
+import {useInViewEffect} from '../../../src/preact/use-in-view';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function Timeago(props) {
+  const {0: timestamp, 1: setTimestamp} = useState(
+    getFuzzyTimestampValue(props)
+  );
+
+  const ref = useRef(null);
+  useInViewEffect(
+    ref.current,
+    () => {
+      setTimestamp(getFuzzyTimestampValue(props));
```

Comment:
Btw, to stress where these things start to compound. In this code, we will have the following sequence:

1. Render 1. The very first one.
2. `useInViewEffect:useEffect:1` -> `useInViewEffect:setState(isIntersecting)`
3. Render 2. This is due to `setState(isIntersecting)`.
4. `useInViewEffect:useEffect:2` will call this method as an effect.
5. This method calls `setTimestamp` state.
6. Render 3. Due to the `setTimestamp`.

On the surface, here we will have only one extra (unnecessary) rerender 2: it's fully unobservable to the component that's using the `useInViewEffect` hook. However, these do add up if every such effect causes an extra rerendering.